components/cdc/src/delegate.rs

```diff
@@ -191,7 +197,14 @@ impl Downstream {
         }
     }
 
+    /// EventErrors must be sent by this function. And we must ensure no more
+    /// events or ResolvedTs will be sent to the downstream after
+    /// `sink_error_event` is called.
     pub fn sink_error_event(&self, region_id: u64, err_event: EventError) -> Result<()> {
+        info!("cdc downstream meets region error";
+            "conn_id" => ?self.conn_id, "downstream_id" => ?self.id, "req_id" => self.req_id);
+
+        self.scan_truncated.store(true, Ordering::Release);
```

Comment:
Will there be any race condition that some unexpected scaned entries have been send to TiCDC before the flag is set ?

Reply:
It's OK. We just need to ensure no more events will be sent to the downstream **after the `EventError` is drained**.